pkg/controllers/subnetset/subnetset_webhook.go

```diff
@@ -0,0 +1,90 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package subnetset
+
+import (
+	"context"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+// log is for logging in this package.
+var subnetsetlog = logf.Log.WithName("subnetset-webhook")
+
+var NSXOperatorSA = "system:serviceaccount:vmware-system-nsx:nsx-operator-svc-account"
+
+// Create validator instead of using the existing one in controller-runtime because the existing one can't
+// inspect admission.Request in Handle function.
+
+//+kubebuilder:webhook:path=/validate-nsx-vmware-com-v1alpha1-subnetset,mutating=false,failurePolicy=fail,sideEffects=None,groups=nsx.vmware.com.nsx.vmware.com,resources=subnetsets,verbs=create;update,versions=v1alpha1,name=vsubnetset.kb.io,admissionReviewVersions=v1
+
+type SubnetSetValidator struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+func isDefaultSubnetSet(s *v1alpha1.SubnetSet) bool {
```

Comment:
this cannot forbid the case when normal user creates a SubnetSet with "DefaultPodSubnetSet" or 'DefaultVMSubnetSet" label, right?

Reply:
I think it can do this. In function `isDefaultSubnetset`, you can see that a SubnetSet with "DefaultPodSubnetSet" or 'DefaultVMSubnetSet" label will be considered as default subnetset. So that webhook will deny the creation from normal user.